counsel-gtags.el

```diff
@@ -81,166 +87,114 @@ searching for a tag."
   "List of extra arguments passed to gtags when updating database."
   :type 'list)
 
-(defcustom counsel-gtags-prefix-key "\C-c"
+(defcustom counsel-gtags-prefix-key (kbd "C-c g")
   "Key binding used for `counsel-gtags-mode-map'.
 This variable does not have any effect unless
 `counsel-gtags-use-suggested-key-map' is non-nil."
   :type 'string)
 
-(defvaralias 'counsel-gtags-suggested-key-mapping 'counsel-gtags-use-suggested-key-map)
-(defcustom counsel-gtags-use-suggested-key-map nil
-  "Whether to use the suggested key bindings."
+(defcustom counsel-gtags-use-suggested-key-map t
+  "Whether to use the suggested key bindings.
+This variable must be set before enabling the mode"
   :type 'boolean)
-(make-obsolete-variable 'counsel-gtags-suggested-key-mapping 'counsel-gtags-use-suggested-key-map "0.01")
 
 (defconst counsel-gtags--prompts
   '((definition . "Find Definition: ")
-    (reference  . "Find Reference: ")
+    (file      . " Find File: ")
     (pattern    . "Find Pattern: ")
+    (reference  . "Find Reference: ")
     (symbol     . "Find Symbol: ")))
 
 (defconst counsel-gtags--complete-options
-  '((file      . "-P")
+  '((definition . "-d")
+    (file      . "-P")
     (pattern   . "-g")
     (reference . "-r")
     (symbol    . "-s")))
 
 (defvar counsel-gtags--last-update-time 0)
 (defvar counsel-gtags--context nil)
-(defvar counsel-gtags--context-position 0)
+(defvar counsel-gtags--other-window nil
+  "Helper global variable to implement other window functions.")
```

Comment:
This is a helper variable, nobody must touch it look how it is used.

Reply:
I was thinking of expanding on the docstring because it would help other _developers_ and not end users of this package.

What I'm trying to convey is that I only understood the need for this variable after reading through the rest of the changes ("is it going to be `setq-local`? must it? is `nil` a valid value or should it be initialized on some event that's expected to happen?").

Adding this information as docstrings helps maintainability since the developer can have immediate access to this type of information and do a mental map of how the events happen, how the code is executed and contracts before / after function calls.